bddisasm/bdx86_decoder.c

```diff
@@ -4427,11 +4432,19 @@ NdDecodeWithContext(
     Instrux->EncMode = ND_ENCM_LEGACY;  // Assume legacy encoding by default.
 
     // Fetch the instruction bytes.
-    for (opIndex = 0; 
-         opIndex < ((Size < ND_MAX_INSTRUCTION_LENGTH) ? Size : ND_MAX_INSTRUCTION_LENGTH); 
-         opIndex++)
+    if (Size < ND_MAX_INSTRUCTION_LENGTH)
```

Comment:
This just moves the condition from the `for` outside. Modern compilers should be able to deal with it as it is (for example, MSVC unrolls the entire loop, so introducing this `if` actually makes things worse).

Reply:
This is very likely GCC doing something stupid. I'll test clang too and see what happens.